subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -188,12 +254,114 @@ static int prepare_cb(struct lll_prepare_param *p)
 	/* Calculate the current event counter value */
 	event_counter = lll->event_counter + lll->skip_event;
 
+	/* Reset accumulated latencies */
+	lll->skip_prepare = 0;
+
+	chan_idx = data_channel_calc(lll);
+
 	/* Update event counter to next value */
 	lll->event_counter = (event_counter + 1);
 
+	err = prepare_cb_common(p, chan_idx);
+	if (err) {
+		DEBUG_RADIO_START_O(1);
+
+		return 0;
+	}
+
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
+	struct lll_df_sync_cfg *cfg;
+
+	cfg = lll_df_sync_cfg_latest_get(&lll->df_cfg, NULL);
+
+	if (cfg->is_enabled) {
+		lll_df_conf_cte_rx_enable(cfg->slot_durations, cfg->ant_sw_len, cfg->ant_ids,
+					  chan_idx);
+		cfg->cte_count = 0;
+
+		radio_switch_complete_and_phy_end_disable();
+	} else
+#endif /* CONFIG_BT_CTLR_DF_SCAN_CTE_RX */
+	{
+		radio_df_cte_inline_set_enabled(false);
+		radio_switch_complete_and_disable();
+	}
+
+	radio_isr_set(isr_rx_adv_sync_estab, lll);
+
+	DEBUG_RADIO_START_O(1);
+
+	return 0;
+}
+
+static int prepare_cb(struct lll_prepare_param *p)
+{
+	uint16_t event_counter;
+	struct lll_sync *lll;
+	uint8_t chan_idx;
+	int err;
+
+	DEBUG_RADIO_START_O(1);
+
+	lll = p->param;
+
+	/* Calculate the current event latency */
+	lll->skip_event = lll->skip_prepare + p->lazy;
+
+	/* Calculate the current event counter value */
+	event_counter = lll->event_counter + lll->skip_event;
```

Comment:
I realize that this function has just been moved and this is outside the scope of this PR, but is there any chance that these additions could cause an overflow? We are adding two `uint16_t`s.

Reply:
Could you add a comment regarding that here?

Reply:
Yyy... comment regarding what? There is a response above, did I missed something?

Reply:
I meant adding a comment in the code, stating that the event_counter may overflow, but that is it okay, as it can wrap.